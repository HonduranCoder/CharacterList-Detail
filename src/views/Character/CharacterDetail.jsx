import React from 'react';
import { useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';
import { getCharacter } from '../../services/airbender.js';
import Character from '../../components/Characters/Character.jsx';

export default function CharacterDetail() {
  const { id } = useParams();
  const [character, setCharacter] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const newCharacter = async () => {
      const returnChar = await getCharacter(id);
      setCharacter(returnChar);
      setLoading(false);
    };
    newCharacter();
  }, [id]);

  return (
    <>
      <h1>Character Detail</h1>
      {loading ? <h3>Loading</h3> : <Character character={character} />}
      <Link to="/characterlist">Back to Characters</Link>
    </>
  );
}
