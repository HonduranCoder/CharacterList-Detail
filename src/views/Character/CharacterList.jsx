import React from 'react';
import { useEffect, useState } from 'react';
import Characters from '../../components/Characters/Characters.jsx';
import { getCharacters } from '../../services/airbender.js';

export default function CharacterList() {
  const [characters, setCharacters] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const allCharacters = async () => {
      const returnChars = await getCharacters();
      setCharacters(returnChars);
      setLoading(false);
    };
    allCharacters();
  }, []);

  return (
    <div>
      <h1>Character List</h1>
      <div>
        {loading ? <h3>Loading</h3> : <Characters characters={characters} />}
      </div>
    </div>
  );
}
