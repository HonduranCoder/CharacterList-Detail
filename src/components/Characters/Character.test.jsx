import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Character from './Character.jsx';

it('render the character view', () => {
  const character = {
    _id: '5cf5679a915ecad153ab68c8',
    position: 'Earth King',
    allies: 'Roayl Earthbender Guards',
  };

  const { container } = render(
    <MemoryRouter>
      <Character character={character} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
