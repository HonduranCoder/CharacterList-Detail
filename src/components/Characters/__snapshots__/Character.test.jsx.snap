// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the character view 1`] = `
<div>
  <img
    width="500"
  />
  <div>
    name: 
  </div>
  <div>
    position: 
    Earth King
  </div>
  <div>
    allies: 
    Roayl Earthbender Guards
  </div>
</div>
`;
